tests/components/fritzbox/test_config_flow.py

```diff
@@ -165,15 +174,34 @@ async def test_ssdp_already_in_progress(hass: HomeAssistantType, fritz: Mock):
     assert result["reason"] == "already_in_progress"
 
 
+async def test_ssdp_already_in_progress_host(hass: HomeAssistantType, fritz: Mock):
+    """Test starting a flow from discovery twice."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "ssdp"}, data=MOCK_SSDP_DATA
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "confirm"
+
+    MOCK_NO_UNIQUE_ID = MOCK_SSDP_DATA.copy()
+    del MOCK_NO_UNIQUE_ID[ATTR_UPNP_UDN]
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "ssdp"}, data=MOCK_NO_UNIQUE_ID
+    )
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_in_progress"
+
+
 async def test_ssdp_already_configured(hass: HomeAssistantType, fritz: Mock):
     """Test starting a flow from discovery when already configured."""
     result = await hass.config_entries.flow.async_init(
         DOMAIN, context={"source": "user"}, data=MOCK_USER_DATA
     )
     assert result["type"] == "create_entry"
+    assert not result["result"].unique_id
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5422434)